catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/MainMenuFragment.java

```diff
@@ -157,10 +164,22 @@ public void onItemClick(@ButtonId int id) {
 				startActivity(new Intent(getActivity(), WebViewActivity.class));
 				break;
 			case UPLOAD:
-				setShowProgressBar(true);
-				Intent intent = new Intent(getActivity(), ProjectUploadActivity.class)
-						.putExtra(ProjectUploadActivity.PROJECT_NAME, Utils.getCurrentProjectName(getActivity()));
-				startActivity(intent);
+				Project currentProject;
```

Comment:
please move to line `170` so that:
``` Project currentProject = XstreamSerializer.getInstance().loadProject(currentProjectName, getActivity());```

I know I'm especially annoying with this PR. :crying_cat_face: